src/storage/src/mut_records.rs

```diff
@@ -98,9 +122,40 @@ impl MutFileRecords {
         if self.f_sink.can_be_appended(buffer.len() as u64) {
             debug!("writing {} bytes at: {}", buffer.len(), self.path.display());
             self.f_sink.write_all(&buffer).await?;
-            // for now, we flush for every send
-            self.f_sink.flush().await?;
-            debug!("flushed {}", self.path.display());
+            match self.flush_policy {
+                NoFlush => {}
+
+                EveryWrite => {
+                    self.f_sink.flush().await?;
+                    debug!("EveryWrite flushed {}", self.path.display());
+                }
+
+                CountWrites {
+                    n_writes: n,
+                    count: icounts,
+                } => {
+                    let cnt = icounts + 1;
+                    if cnt >= n {
+                        self.flush_policy = CountWrites {
```

Comment:
Few questions
1) Shouldn't count be updated after flushing?
2) What happens flushing fails?
3) Instead of creating new CountWrite.  Should just ask CountWrite to update.

Reply:
The counts were updated the logic before, but the new commit commit increments just the field in place instead.
If the flush fails, it should be reported upwards as a send error Result